shared/actions/fs/index.js

```diff
@@ -783,6 +787,79 @@ const cancelMoveOrCopy = isMobile
   ? (state, action) => Saga.put(switchTo([Tabs.settingsTab, SettingsConstants.fsTab, 'folder']))
   : (state, action) => Saga.put(navigateUp())
 
+const showSendLinkToChat = (state, action) => {
+  const elems = Types.getPathElements(state.fs.sendLinkToChat.path)
+  const routeChange = Saga.put(
+    action.payload.routePath
+      ? putActionIfOnPath(action.payload.routePath, navigateAppend(['sendLinkToChat']))
+      : navigateAppend(['sendLinkToChat'])
+  )
+  if (elems.length < 3) {
+    // Not a TLF; so just show the modal and let user copy the path.
+    return routeChange
+  }
+
+  const actions = [routeChange]
+
+  if (elems[1] !== 'team') {
+    // It's an impl team conversation. So resolve to a convID directly.
+    actions.push(
+      Saga.callUntyped(function*() {
+        const result = yield Saga.callPromise(RPCChatTypes.localFindConversationsLocalRpcPromise, {
+          identifyBehavior: RPCTypes.tlfKeysTLFIdentifyBehavior.chatGui,
+          membersType: RPCChatTypes.commonConversationMembersType.impteamnative,
+          oneChatPerTLF: false,
+          tlfName: elems[2],
+          topicName: '',
+          topicType: RPCChatTypes.commonTopicType.chat,
+          visibility: RPCTypes.commonTLFVisibility.private,
+        })
+
+        if (!result.conversations || !result.conversations.length) {
+          // TODO: error?
+          return
+        }
+
+        yield Saga.put(
+          FsGen.createSetSendLinkToChatConvID({
+            convID: ChatTypes.conversationIDToKey(result.conversations[0].info.id),
+          })
+        )
+      })
+    )
+  } else {
+    // It's a real team, but we don't know if it's a small team or big team. So
+    // fire TeamsGen.getChannels to find out. We'll listen for
+    // TeamsGen.setTeamChannels and see if we should automatically set convID.
+    // We might already have that, but just refresh anyway to get latest info.
+    actions.push(Saga.put(TeamsGen.createGetChannels({teamname: elems[2]})))
```

Comment:
i worry about this type of cross store logic. we dispatch some foreign action and then reach into that store to get the value. why not just make the rpcs calls you need and have it be self contained